src/iter/test.rs

```diff
@@ -1608,3 +1608,14 @@ fn check_partition_map() {
     assert_eq!(a, vec![1, 2, 3]);
     assert_eq!(b, "abcxyz");
 }
+
+#[test]
+fn check_repeat_eq() {
+    let v = vec!(4,4,4,4);
+    let mut fours: Vec<_> = repeat(4)
+    .take(4)
+    .zip(v)
+    .collect();
```

Comment:
Yes, exactly.  We can't use this as an `IndexedParallelIterator` unless we actually map it to indexes, which puts an upper bound on the length.

Well... actually if you want to be clever, you could track the length as `Option<usize>` and leave the rightmost part of the splits open-ended.  So a bounded segment with `Some(len)` splits into two bounded segments, and an unbounded `None` splits into `Some` on the left and `None` on the right.  This will still work with `take` because it splits and discards the (infinite) remainder.